### PR TITLE
[HttpFoundation][Cookie] Cookie DateTimeInterface fix

### PR DESCRIPTION
**Pull request by wildewouter**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

I came across an issue with expiration times on cookies. They were not working with DateTimeImmutable but only the DateTime implementation itself. I refactored this to work with the DateTimeInterface.

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Cookie.php [hunk 1]
```diff
@@ -51,7 +51,7 @@ public function __construct($name, $value = null, $expire = 0, $path = '/', $dom
         }
 
         // convert expiration time to a Unix timestamp
-        if ($expire instanceof \DateTime) {
+        if ($expire instanceof \DateTimeInterface) {
```
Comment (Tobion) [inline src/Symfony/Component/HttpFoundation/Cookie.php]: this does not work on PHP 5.3

**Comment by wildewouter** [comment]
Closed in favor of https://github.com/symfony/symfony/pull/17371

**Comment by xabbuh** [comment]
Well, we could still additionally check for `DateTimeImmutable` instances on older Symfony versions.

File: src/Symfony/Component/HttpFoundation/Cookie.php (hunk 1 above)
Comment (Tobion) [inline src/Symfony/Component/HttpFoundation/Cookie.php]: instanceof \DateTime || instanceof \DateTimeInterface

File: src/Symfony/Component/HttpFoundation/Tests/CookieTest.php [hunk 2]
```diff
@@ -85,6 +85,14 @@ public function testConstructorWithDateTime()
         $this->assertEquals($expire->format('U'), $cookie->getExpiresTime(), '->getExpiresTime() returns the expire date');
     }
 
+    public function testConstructorWithDateTimeImmutable()
```
Comment (Tobion) [inline src/Symfony/Component/HttpFoundation/Tests/CookieTest.php]: `@requires PHP 5.5` annotation

**Comment by xabbuh** [comment]
:+1: 

Status: Reviewed

**Comment by fabpot** [comment]
Thank you @wildewouter.

**Pull request closed without merging**